### DISCUSSION TITLE
如何在nuxt3中自定义路由或者说如何借用现在的文件式路由满足我的需求

### DISCUSSION THREAD

**Discussion by zhaohg1** [nuxt/nuxt#28874] [General]
有人说建立一个app文件然后新建什么router.option.js文件我试了，报错，是不是我还要配置什么内容哟，我是直接把nuxt.config.ts中的pages置为false了。我的目的是想每个路由配置一些常量参数用于判断每个路由是否需要登录才能访问，我不想每个vue文件都去里面写这个判断逻辑，路由太多了，不同路由的情况不同，有很多配置参数，vue2的时候可以配置路由参数meta，现在nuxt3怎么做到这个概念哟

**Comment by xjccc** [comment]
You could set [definePageMeta](https://nuxt.com/docs/api/utils/define-page-meta), and access `route.meta`
```
definePageMeta({
    ...your params
})
```
